### ISSUE TITLE
CustomNonbondedForce.addInteractionGroup should accept numpy integer types as well

### ISSUE THREAD

**Issue by jchodera** [openmm/openmm#1650]
When calling `addInteractionGroup`, adding `list`s or `set`s works fine:

```
>>> import numpy as np
>>> from simtk import openmm
>>> force = openmm.CustomNonbondedForce('1')
>>> force.addInteractionGroup([1,2,3], [4,5,6])
0
```

but if the list contains `numpy` integer types, this fails:

```
>>> force.addInteractionGroup(np.array([1,2,3], np.int64), [4,5,6])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/cbio/jclab/home/chodera/miniconda3/envs/sams35/lib/python3.5/site-packages/simtk/openmm/openmm.py", line 9950, in addInteractionGroup
    return _openmm.CustomNonbondedForce_addInteractionGroup(self, set1, set2)
TypeError: in method 'CustomNonbondedForce_addInteractionGroup', argument 2 of type 'std::set< int,std::less< int >,std::allocator< int > > const &'
```

It would be convenient to accept `numpy` integer types as well.

**Comment by peastman** [comment]
Try changing the `int64` to `int32`.  The problem may just be that you're passing the wrong data type.  The argument to the C++ method is a `const set<int>&`, so it ought to accept anything that can be interpreted as a sequence of 32 bit integers.

**Comment by jchodera** [comment]
Good idea, but looks like that doesn't work:
```
>>> import numpy as np
>>> from simtk import openmm
>>> force = openmm.CustomNonbondedForce('1')
>>> force.addInteractionGroup(np.array([1,2,3], np.int32), [4,5,6])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/cbio/jclab/home/chodera/miniconda3/envs/sams35/lib/python3.5/site-packages/simtk/openmm/openmm.py", line 9950, in addInteractionGroup
    return _openmm.CustomNonbondedForce_addInteractionGroup(self, set1, set2)
TypeError: in method 'CustomNonbondedForce_addInteractionGroup', argument 2 of type 'std::set< int,std::less< int >,std::allocator< int > > const &'
```
We ran into this issue because `mdtraj` apparently uses a `numpy.int64` array to return atom selections.

**Comment by peastman** [comment]
I don't have any other suggestions then.  The support for STL templates with standard types is all built into SWIG.  It's *supposed* to accept any sequence type that contains ints.

**Comment by jchodera** [comment]
I've been able to work around this with the idiom
```python
indices = [ int(index) for index in input_indices ]
```
but I'm not sure whether it would be dangerous to build that in as a fallback case or not.

**Comment by peastman** [comment]
A simpler way is

```Python
indices = list(input_indices)
```

> I'm not sure whether it would be dangerous to build that in as a fallback case or not.

I'm not even sure how we *could* do that, short of ignoring the built in typemap and implementing our own from scratch.  And a better option might be to just report this to the SWIG developers as a bug.

**Comment by jchodera** [comment]
Are we using [`numpy.i`](https://docs.scipy.org/doc/numpy/reference/swig.interface-file.html)?

Another clue: This issue only appeared on python 3.x in which `numpy 1.11` was installed. Perhaps our support for `numpy 1.11` is incomplete? We're currently building with `swig 3.0.8`, which I believe predates `numpy 1.11` by at least three months.

**Comment by rmcgibbo** [comment]
No, we don't use `numpy.i`.

**Comment by peastman** [comment]
This works in the latest code.  I think it was probably fixed by #2961.
